pkg/exporter/azureblob_exporter.go

```diff
@@ -76,6 +76,24 @@ func createContainerURL() (azblob.ContainerURL, error) {
 	return containerURL, nil
 }
 
+// GetStorageContainerName get storage container name
+func getStorageContainerName(APIServerFQDN string) (string, error) {
+	containerName := strings.Replace(APIServerFQDN, ".", "-", -1)
+
+	//TODO DK: I really dont like the line below, it makes for weird behaviour if e.g. .hcp. or -hcp- is in the fqdn for some reason other than being auto-added by AKS
+	length := strings.Index(containerName, "-hcp-")
+
+	if length == -1 {
+		maxLength := len(containerName)
+		length = int(math.Min(float64(maxLength), float64(maxContainerNameLength)))
```

Comment:
💡 What is this logic? not able to follow this. why const is `float64` ? Why are we finding min at this point is not clear either, (Is this replacement of any old logic if you can point me to please)

Reply:
this replaces the code in lines 41-46 of master. 

No function signature overloading in Go means math.Min is only defined for float64 and we have to pollute with casts like this. Alternative is probably something like adding an explicit "MinInt" func.

We need to get the min to enforce 63 char limit, rest of the logic does stuff needed to make the test cases pass :)

At some point its worth refactoring the -hcp- check to give less "false positives" by pattern matching only to strings ending with "hcp.<region>.azmk8s.io"